### ISSUE TITLE
`buttons.css` should include classnames

### ISSUE THREAD

**Issue by argyleink** [argyleink/open-props#310]
add `.button`, `.primary`, `.light`, `.dark` to the buttons file and add examples of usage on the [demo page](https://codepen.io/argyleink/pen/gOXbvjj)

**Comment by WebMechanic** [comment]
May I suggest to use/add **visually neutral** names like `.primary`, `.secondary`, and `.accent` that could eventually define whatever colour theme a site's using. In some areas we found these thee _meaningful_ major colours need a _neutral_ companion and added `.compl` for the then _visually complementary_ stuff (buttons, badges, icons).

`.light` and `.dark` only make sense if all colour themes allow for such a visual concept. What shade does the `.dark` button of a dark theme have if the primary is dark already? 
A secondary button or element however could be either _dark_ or _light_ based off the CD and the active colour theme/context.

Our version of `.button` is also used to style anything that isn't a real `<button>` to look like one (+ JS for the fun) and `.link-like` to make real buttons look like anchors.

For our CTA set we also added "modifiers" `.-ghost` and `.-contour` that _modify_ button fill, borders and outlines, but take the colours from whatever is defined by one of the mutually exclusive main classes (primary, secondary, accent).

This quartet is also applied for other coloured things to carry meaning or purposes of an element or area (--surface-secondary).

<details><summary> Button Modifiers </summary>
They're almost utility classes but with a stricter scope.

```css
.btn {
  &.-primary {
	--btn-shade: var(--brand-primary-100);
	--btn-base: var(--brand-primary-400);
	--btn-border: var(--brand-primary-400);
	--btn-hover: var(--brand-primary-600);
	--btn-text: var(--color-10);
	--btn-text-hover: var(--color-10);
  }
  &.-secondary {
	--btn-shade: var(--brand-secondary-100);
	--btn-base: var(--brand-secondary-400);
	--btn-hover: var(--brand-secondary-600);
	--btn-text: var(--color-10);
	--btn-text-hover: var(--btn-shade);
  }
  &.-accent {
	--btn-shade: var(--brand-accent-100);
	--btn-base: var(--brand-accent-400);
	--btn-hover: var(--brand-accent-600);
	--btn-text: var(--color-10);
	--btn-text-hover: var(--btn-shade);
  }
  &.-compl {
	--btn-shade: var(--brand-compl-100);
	--btn-base: var(--brand-compl-400);
	--btn-hover: var(--brand-compl-600);
	--btn-text: var(--color-10);
	--btn-text-hover: var(--btn-shade);
  }
  &:not(.-primary) {
	--btn-border: var(--btn-base);
  }
  &.-ghost {
	--btn-base: var(--surface-10);
	--btn-border: var(--btn-text);
  }
  &.-big { ... }
  &.-half { ... }
}
```
As you can see they're actually all "modifier classes" that don't work on their own and each will change different properties. They _must_ be used with `.btn` but can be combined. 
There can always be a primary, secondary or accented "thing", i.e. for the card component we use the aptly named `.card` to define their basic look and there are equally named modifiers to make adjustments where we see fit.

You can throw in any colours you want (-100 = darkest, -900 = lightest)
</details>

**Comment by argyleink** [comment]
One of the ideas of this was to allow specifying the buttons to be only dark or only light, but i hear ya about neutral naming!

this issue will need to create an entirely different button stylesheet for import, that's just utilities 👍🏻  thanks for sharing y'all's process and setup. looks pretty standard (i mean that in a good way!). can you believe some people think buttons are easy? 😂

**Comment by WebMechanic** [comment]
not only are buttons easy, they also work wonders: need a new app or site feature? just slap "a button" on it and you're done (according to user's feature requests). 🦄✨

> the ideas of this was to allow specifying the buttons to be only dark or only light, 

sure, hence:

> What shade does the `.dark` button of a dark theme have ... ?

The concept of "dark" and "light" only works with specific themes / environments. One will soon end up with something called dark that eventually isn't.
That's why I suggested the generic _accent_ and _complementary_ variants :-)

For that very reason I also abandoned classes like "red" or "blue" +20 years ago, even if these would've been the main two corporate colours -- and dark-themes weren't really a thing then, but maybe print in black and white.

There'll always be some situation in any design job where identifying sth. by colour name/shade only is virtually _meaningless_ or even _obstructive_ in communication.

Happy New Year! 🍾
